src/test/java/com/google/neighborgood/NotificationServletTest.java

```diff
@@ -0,0 +1,224 @@
+   * that the user is logged in as admin, has a specific email and userID.
+   */
+  private LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(
+              new LocalDatastoreServiceTestConfig(), new LocalUserServiceTestConfig())
+          .setEnvIsAdmin(true)
+          .setEnvIsLoggedIn(true)
+          .setEnvEmail("leonardzhang@google.com")
+          .setEnvAuthDomain("1234567890")
+          .setEnvAttributes(
+              ImmutableMap.of(
+                  "com.google.appengine.api.users.UserService.user_id_key", "1234567890"));
+
+  private UserService userService;
+  private DatastoreService ds;
+  private HttpServletRequest request;
+  private HttpServletResponse response;
+  private Entity userEntity;
+  private Entity taskEntity;
+  private String keyString;
+  private String anotherKey;
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+    userService = UserServiceFactory.getUserService();
+    ds = DatastoreServiceFactory.getDatastoreService();
+    request = mock(HttpServletRequest.class);
+    response = mock(HttpServletResponse.class);
+
+    userEntity = new Entity("UserInfo", "1234567890");
+    userEntity.setProperty("nickname", "Leonard");
+    userEntity.setProperty("address", "xxx");
+    userEntity.setProperty("email", "leonardzhang@google.com");
+    userEntity.setProperty("userId", "1234567890");
+    userEntity.setProperty("country", "US");
+    userEntity.setProperty("zipcode", "15213");
+    userEntity.setProperty("points", 0);
+    ds.put(userEntity);
+
+    taskEntity = new Entity("Task", userEntity.getKey());
+    taskEntity.setProperty("detail", "Test task");
+    taskEntity.setProperty("timestamp", 123);
+    taskEntity.setProperty("reward", 50);
+    taskEntity.setProperty("status", "IN PROGRESS");
+    taskEntity.setProperty("Owner", "1234567890");
+    taskEntity.setProperty("Helper", "1234567890");
+    taskEntity.setProperty("Address", "xxx");
+    taskEntity.setProperty("zipcode", "15213");
+    taskEntity.setProperty("country", "US");
+    taskEntity.setProperty("category", "Garden");
+    ds.put(taskEntity);
+
+    keyString = KeyFactory.keyToString(taskEntity.getKey());
+
+    taskEntity = new Entity("Task", userEntity.getKey());
+    taskEntity.setProperty("detail", "Test task 2");
+    taskEntity.setProperty("timestamp", 123);
+    taskEntity.setProperty("reward", 50);
+    taskEntity.setProperty("status", "IN PROGRESS");
+    taskEntity.setProperty("Owner", "1234567890");
+    taskEntity.setProperty("Helper", "1234567890");
+    taskEntity.setProperty("Address", "xxx");
+    taskEntity.setProperty("zipcode", "15213");
+    taskEntity.setProperty("country", "US");
+    taskEntity.setProperty("category", "Garden");
+    ds.put(taskEntity);
+
+    anotherKey = KeyFactory.keyToString(taskEntity.getKey());
+
+    // Add 3 dummy notification entities with the current user as receiver and taskId equals to
```

Comment:
I really like all the comments you put in this test! It makes it much easier to understand 😄

Reply:
Thanks! 😄